design/milestone_1_2_design_doc.md

```diff
@@ -350,25 +350,27 @@ For a detailed breakdown of the decision process see [here](batch_retrieval_desi
-| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
-| 1    | Every configurable amount of time, retrieve secrets <br />*Default: 5 minutes (300 seconds)* | - Every 5 minutes, perform Secrets Provider cycle <br />- Fetch Conjur secrets and update K8s Secrets | - Easy integration with current retry mechanism<br />- No need to make a change in Conjur<br />- No changes in  Helm Charts between Milestones | - Polling is a bad way for a change notification mechanism<br />- Load on the server every 5 minutes | 5 days            |
-| 2a   | Every configurable amount of time, check if Conjur secret was updated | - Create a new API endpoint to check if Conjur secret version was updated<br />Return 200 status code<br />- Trigger Secrets Provider cycle on all K8s Secrets with values in Conjur | - Less load on Conjur server<br />- No changes in Helm Chart between Milestones | - New API endpoint is needed in Conjur<br />- API endpoint will *not* return which secret was updated<br />- Users will need to update their Conjur versions to use the endpoint (*) | 10 days           |
-| 2b   | Every configurable amount of time, check if Conjur secret was updated | - Expand on the current Batch Retrieval API to return list of secrets and **also** **versions**<br /><br />- If newer version is returned, trigger  Secrets Provider cycle only with the returned list of updated Conjur secrets | - Don't need to create a new API                             | - Complex solution<br />- We need to ensure we are not introducing breaking changes | 15 days           |
+|      | Solution                               | Elaboration                                                  | Pros                                                         | Cons                                                         | Effort estimation |
+| ---- | -------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Client-side polling for secrets        | Every configurable amount of time, contact Conjur API server.<br />Possible solutions can be:<br />1. Retrieve all secrets<br />2. Check if there's something new for the requesting host<br />3. Retrieve secrets that were changed since last request | - Straight-forward<br />- Simple<br />- No additional requirements needed | - Regularly sends API calls even if nothing has changed<br />- Introduces load on the server to hanlde many secrets providers | 15 days           |
```

Comment:
Unclear what is the "something new"
> Check if there's a secret update for the requesting host

Reply:
what does "No additional requirements needed" mean?

Reply:
>  Introduces load on the server to hanlde many secrets providers

Load on server to handle additional requests, especially with many secrets providers